### PR TITLE
Refactor TPOT so each operator is in its own class

### PR DESCRIPTION
**Pull request by teaearlgraycold**
Even though this isn't finished I figured I may as well put this up here as a PR.

Currently this is what the exported code looks like:

```Python
import pandas as pd

from sklearn.cross_validation import train_test_split
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import MinMaxScaler
from sklearn.svm import LinearSVC

# NOTE: Make sure that the class is labeled 'class' in the data file
tpot_data = pd.read_csv('PATH/TO/DATA/FILE', sep='COLUMN_SEPARATOR')
training_indices, testing_indices = train_test_split(tpot_data.index, stratify=tpot_data['class'].values, train_size=0.75, test_size=0.25)

exported_pipeline = Pipeline([
    ("0-Preprocessor", MinMaxScaler(copy=False)),
    ("1-Classifier", LinearSVC(dual=True, C=0.1, penalty="l2"))
])

exported_pipeline.fit(tpot_data.loc[training_indices].drop('class', axis=1).values,
                      tpot_data.loc[training_indices, 'class'].values)
results = exported_pipeline.predict(tpot_data.loc[testing_indices].drop('class', axis=1))
```

- [X] Each TPOT operator is in a class
- [X] TPOT exports code to an sklearn pipeline
- [X] Pipelines are restricted so that the last operator is always a classifier
- [X] Finish rewriting tests
- [X] Discuss/rework pipeline structure (so far from what I've seen the pipeline structure enforcing has shown accuracy scores that are at least equal with the current 0.4.1 scores)
- [x] Remove pandas as a dependancy
- [ ] Update docs/tutorials

### CODE REVIEW

**Comment by teaearlgraycold** [comment]
Coverage is now yellow!

Now just need to write some more tests so the coverage actually means something

**Comment by teaearlgraycold** [comment]
So this is what we're looking at now with exported code:

```Python
import numpy as np
import pandas as pd

from sklearn.cross_validation import train_test_split
from sklearn.ensemble import GradientBoostingClassifier, VotingClassifier
from sklearn.feature_selection import RFE
from sklearn.neighbors import KNeighborsClassifier
from sklearn.pipeline import make_pipeline, make_union
from sklearn.preprocessing import FunctionTransformer
from sklearn.svm import SVC

# NOTE: Make sure that the class is labeled 'class' in the data file
tpot_data = pd.read_csv('PATH/TO/DATA/FILE', sep='COLUMN_SEPARATOR')
training_indices, testing_indices = train_test_split(tpot_data.index, stratify=tpot_data['class'].values, train_size=0.75, test_size=0.25)

exported_pipeline = make_pipeline(
    make_union(
        make_union(VotingClassifier(estimators=[('branch',
            GradientBoostingClassifier(learning_rate=1.0, max_features=1.0, min_weight_fraction_leaf=0.5, n_estimators=500)
        )]), FunctionTransformer(lambda X: X)),
        RFE(estimator=SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
          decision_function_shape=None, degree=3, gamma='auto', kernel='linear',
          max_iter=-1, probability=False, random_state=42, shrinking=True,
          tol=0.001, verbose=False), step=0.18)
    ),
    KNeighborsClassifier(n_neighbors=5, weights="distance")
)

exported_pipeline.fit(tpot_data.loc[training_indices].drop('class', axis=1).values,
                      tpot_data.loc[training_indices, 'class'].values)
results = exported_pipeline.predict(tpot_data.loc[testing_indices].drop('class', axis=1))
```

**Comment by teaearlgraycold** [comment]
This build will still fail the tests but I guess you can go ahead and merge it into the dev branch.
